features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/RingBufferTimeseriesWriter.java

```diff
@@ -187,28 +191,37 @@ public void destroy() {
 
     @Override
     public void insert(List<Sample> samples) {
+        if (!readyToRockAndRoll.get()) {
+            insertDrop(samples, "We are not ready to rock and roll");
+            return;
+        }
 
         // Add the samples to the ring buffer
         if (!ringBuffer.tryPublishEvent(TRANSLATOR, samples)) {
-            RATE_LIMITED_LOGGER.error("The ring buffer is full. {} samples associated with resource ids {} will be dropped.",
-                    samples.size(), new Object() {
-                        @Override
-                        public String toString() {
-                            // We wrap this in a toString() method to avoid build the string
-                            // unless the log message is actually printed
-                            return samples.stream()
-                                    .map(s -> s.getMetric().getFirstTagByKey(IntrinsicTagNames.resourceId).getValue())
-                                    .distinct()
-                                    .collect(Collectors.joining(", "));
-                        }
-                    });
-            droppedSamples.mark(samples.size());
+            insertDrop(samples, "The ring buffer is full");
             return;
         }
         // Increase our entry counter
         numEntriesOnRingBuffer.incrementAndGet();
     }
 
+    private void insertDrop(List<Sample> samples, String message) {
+        RATE_LIMITED_LOGGER.error("{}. {} samples associated with resource ids {} will be dropped.",
```

Comment:
I would suggest changing this to include the `message` in the format string instead of pulling it in with an `{}` anchor. So, change this:
```
    RATE_LIMITED_LOGGER.error("{}. {} samples associated with resource ids {} will be dropped.",
Write
                message,
                samples.size(), new Object() {
```
to this:
```
    RATE_LIMITED_LOGGER.error(message + ". {} samples associated with resource ids {} will be dropped.",
Write
                samples.size(), new Object() {
```

Otherwise, the rate limited log messages aren't great because the message just shows up as `null`, like this:
```
2023-01-23T11:56:39,572 | ERROR | RateLimitedLogRegistry-0 | RingBufferTimeseriesWriter       | 478 - org.opennms.features.org.opennms.features.timeseries - 32.0.0.SNAPSHOT | (suppressed 2 logs similar to 'null. null samples associated with resource ids {} will be dropped.' in PT29.927S)
```

Reply:
Change applied.